parsers/US_PJM.py

```diff
@@ -1,53 +1,60 @@
-#!/usr/bin/env python3
-
-"""Parser for the PJM area of the United States."""
+"""Parser for the PJM area of the United States (US-MIDA-PJM)."""
 
 import re
 from datetime import datetime, time, timedelta, timezone
 from logging import Logger, getLogger
+from typing import Literal
 from zoneinfo import ZoneInfo
 
 import demjson3 as demjson
 import pandas as pd
 from bs4 import BeautifulSoup
-from dateutil import parser
 from requests import Response, Session
 
+from electricitymap.contrib.lib.models.event_lists import (
+    ExchangeList,
+    PriceList,
+    ProductionBreakdownList,
+    TotalConsumptionList,
+)
+from electricitymap.contrib.lib.models.events import (
+    EventSourceType,
+    ProductionMix,
+    StorageMix,
+)
+from electricitymap.contrib.lib.types import ZoneKey
 from parsers.lib.config import refetch_frequency
 from parsers.lib.exceptions import ParserException
 
+PARSER = "US_PJM.py"
 TIMEZONE = ZoneInfo("America/New_York")
-
-# Used for consumption forecast data.
-API_ENDPOINT = "https://api.pjm.com/api/v1/"
-# Used for both production and price data.
-url = "http://www.pjm.com/markets-and-operations.aspx"
-
-mapping = {
-    "Coal": "coal",
-    "Gas": "gas",
-    "Hydro": "hydro",
-    "Multiple Fuels": "unknown",
-    "Nuclear": "nuclear",
-    "Oil": "oil",
-    "Other": "unknown",
-    "Other Renewables": "unknown",
-    "Solar": "solar",
-    "Wind": "wind",
-}
-
-exchange_mapping = {
-    "nyiso": "NYIS|NYIS",
-    "neptune": "NEPTUNE|SAYR",
-    "linden": "LINDENVFT|LINDEN",
-    "hudson": "HUDSONTP|HTP",
-    "miso": "miso",
-    "ohio valley": "DEOK|OVEC",
-    "louisville": "SOUTHIMP|LGEE",
-    "tennessee valley": "SOUTHIMP|TVA",
-    "cpl west": "SOUTHIMP|CPLW",
-    "duke": "SOUTHIMP|DUKE",
-    "cpl east": "SOUTHIMP|CPLE",
+ZONE_KEY = ZoneKey("US-MIDA-PJM")
+
+# Used for production and consumption forecast data (https://dataminer2.pjm.com/list)
+DATA_MINER_API_ENDPOINT = "https://api.pjm.com/api/v1/"
+
+# Used for price data.
+PRICE_API_ENDPOINT = "http://www.pjm.com/markets-and-operations.aspx"
+CURRENCY = "USD"
+
+SOURCE = "pjm.com"
+
+ZONE_TO_PJM_INTERFACES = {
+    ZoneKey("US-MIDW-MISO"): ["MISO"],  # "MISO LMP"
+    # ?: ["DEOK|OVEC"],  # Ohio Valley Electric Corporation (OVEC)
```

Comment:
question: this interface is not shown on the [webpage](https://www.pjm.com/markets-and-operations/interregional-map), but is [technically available as an interface](https://www.pjm.com/Charts/InterfaceChartDM2.aspx?open=DEOK|OVEC) (although it seems to not be returning any data)

Does anyone know what it is / which zone it would connects to?